src/python/pants/backend/project_info/tasks/export_dep_as_jar.py

```diff
@@ -252,67 +257,95 @@ def initialize_graph_info(self, targets_map):
 
     return graph_info
 
-  def generate_targets_map(self, all_targets, runtime_classpath):
+  def _get_all_targets(self, targets):
+    additional_java_targets = []
+    for t in targets:
+      if isinstance(t, ScalaLibrary):
+        additional_java_targets.extend(t.java_sources)
+    targets.extend(additional_java_targets)
+    return targets
+
+  def _get_targets_to_print(self, all_targets, target_roots_set):
+    def target_is_between_roots(t):
+      transitive_dependencies = set(
+        DependencyContext.global_instance().dependencies_respecting_strict_deps(t) if hasattr(t, 'strict_deps') else \
+          DependencyContext.global_instance().all_dependencies(t)
+      )
+      return len(transitive_dependencies.intersection(target_roots_set)) > 0
+
+    def target_is_a_resource(t):
+      return isinstance(t, Resources)
+
+    printable_targets = []
+    for target in all_targets:
+      print(f"BL: COnsidering target {target}, with target_root_set {target_roots_set}. is_a_resource {target_is_a_resource(target)}, is_between_roots {target_is_between_roots(target)}")
+      if (target in target_roots_set) or \
+        target_is_between_roots(target) or \
+        target_is_a_resource(target):
```

Comment:
What is the reason to force resources to be `printable`, i.e. a module in intellij, regardless whether it is a target root? 

Resources in the dependencies are frozen in status quo, and the frozen resources in a jar works as intended already.

Reply:
Removed that constraint :)